embedded/store/key_reader.go

```diff
@@ -19,29 +19,37 @@ import (
 	"bytes"
 	"crypto/sha256"
 	"encoding/binary"
+	"time"
 
 	"github.com/codenotary/immudb/embedded/tbtree"
 )
 
 type Snapshot struct {
 	st             *ImmuStore
 	snap           *tbtree.Snapshot
+	ts             time.Time
```

Comment:
Post-merge: This doesn't look like a part of a snapshot but rather an execution context (which may contain things like snapshot, current timestamp, in the future also locally declared variables for example)

(also this time should be used as a result of the NOW() built-in function so that multiple calls to NOW() function in one statement returns the same value)

Reply:
sure, will continue to be adapted as we continue maturing sql implementation